tool/test.sh

```diff
@@ -0,0 +1,112 @@
+    echo "${fmt_result[@]}" 1>&2
+    exit 1
+  elif [[ "${#fmt_result[@]}" == 0 ]]; then
+    echo "No formatting errors!"
+  else
+    error "There are formatting errors in the following files:"$'\n'
+    for file in "${fmt_result[@]}"; do
+      echo "===== $file =====" 1>&2
+      cp "$file" "$file.expected"
+      "$dartfmt" -w "$file.expected" &> /dev/null
+      echo "----- expected format -----" 1>&2
+      cat "$file.expected" 1>&2
+      echo "----- current format -----" 1>&2
+      cat "$file" 1>&2
+      echo "----- diff -----" 1>&2
+      diff "$file.expected" "$file" 1>&2 || true
+      echo "===== /end $file ====="$'\n' 1>&2
+      rm "$file.expected"
+    done
+    error "Formatting errors detected by dartfmt in ${#fmt_result[@]} dart file(s)."
+    return 1
+  fi
+}
+
+# Deploys website to Firebase, trying up to total_tries times.
+function deploy() {
+  local total_tries="$1"
+  local remaining_tries=$(($total_tries - 1))
+  local project="$2"
+  while [[ "$remaining_tries" > 0 ]]; do
+    # FIREBASE_TOKEN is set in the .cirrus.yml file.
+    firebase deploy --token "$FIREBASE_TOKEN" --non-interactive --project "$project" && break
+    remaining_tries=$(($remaining_tries - 1))
+    error "Error: Unable to deploy project $project to Firebase. Retrying in five seconds... ($remaining_tries tries left)"
+    sleep 5
+  done
+
+  [[ "$remaining_tries" == 0 ]] && {
+    error "Deployment still failed after $total_tries tries: '$@'"
+    return 1
+  }
+  return 0
+}
+
+# Use the version of Dart SDK from the Flutter repository instead of whatever
+# version is in the PATH. FLUTTER_ROOT comes from the Cirrus Docker image.
+export PATH="$FLUTTER_ROOT/bin/cache/dart-sdk/bin:$FLUTTER_ROOT/bin:$PATH"
+
+dart="$FLUTTER_ROOT/bin/cache/dart-sdk/bin/dart"
+dartfmt="$FLUTTER_ROOT/bin/cache/dart-sdk/bin/dartfmt"
+pub="$FLUTTER_ROOT/bin/cache/dart-sdk/bin/pub"
+flutter="$FLUTTER_ROOT/bin/flutter"
+
+echo "Using Dart version:"
+"$dart" --version
+
+"$flutter" packages get
+
+# Analyze the stand-alone sample code files
+for sample in _includes/code/*/*; do
+  if [[ -d "${sample}" ]]; then
+    echo "Run flutter packages get on ${sample}"
+    "$flutter" packages get ${sample}
+  fi
+done
+
+echo "Run flutter analyze on _includes/code/"
+"$flutter" analyze --no-current-package _includes/code/
+
+echo "Extract Dart snippets from the markdown documentation."
+"$dart" tool/extract.dart
+
+echo "Analyzing the extracted Dart libraries."
+"$flutter" analyze --no-current-package example/
+
+echo "Check formatting of the extracted Dart libraries."
+check_formatting example/*.dart
+
+echo "Building site."
+bundle exec jekyll build --verbose
+
+echo "Validating all links."
+rake checklinks
+echo "SUCCESS: Tests passed."
+
+# Deploy on all non-PR master branch builds.
+if [[ -z "$CIRRUS_PR" && "$CIRRUS_BRANCH" == "master" ]]; then
+  echo "Deploying website to Firebase."
+  deploy 5 sweltering-fire-2088
+  echo "SUCCESS: Website deployed successfully."
+fi
```

Comment:
It is possible, but I didn't know the parameters of what that should be (what branches, where to deploy, etc). I removed the commented-out staging server stuff because Filip said it was broken and unused.

I could do it in another PR, but maybe you guys want to do that yourselves to be sure to get what you need?

Reply:
SGTM